### DISCUSSION TITLE
https://theme-hope.vuejs.press/zh/cookbook/tutorial/deploy.html

### DISCUSSION THREAD

**Discussion by giscus** [vuepress-theme-hope/giscus-discussions#566] [Announcements]
# https://theme-hope.vuejs.press/zh/cookbook/tutorial/deploy.html

本教程指引你如何部署 VuePress 项目。

https://theme-hope.vuejs.press/zh/cookbook/tutorial/deploy.html

**Comment by lijianhao0611** [comment]
Error: usePageHead() is called without provider. 构建报错

**Reply by Mister-Hope** [reply]
# READ FAQ

**Comment by huamus** [comment]
部署到github pages这里可以详细写一下吗，没看懂怎么操作

**Reply by Mister-Hope** [reply]
主题会自动生成正确的 workflow，关于工作流是什么，请自行阅读 Github 文档。

**Comment by Oneupupup** [comment]
新建一个仓库名为 `<USERNAME>.github.io` 之后，将所有文件上传到该仓库，上传之后，`github` 的 `action` 会自动部署，但是，需要修改一个地方 setting->Pages->Build and deployment->Branch，将其修改为 `gh-pages`。

**Reply by Mister-Hope** [reply]
开启 Pages 时， Pages 会优先使用 gh-pages，之后是你的默认分支。开启后即使你新拥有了 gh-pages 也不会改变。所以我怀疑你是先开了 Pages 后完成的工作流。

**Comment by theCodedMike** [comment]
"部署到 GitHub Pages"这里能否详细写一下，我按照小白教程走了一遍，部署后，打开https://<USERNAME>.github.io/，此时页面展示的是项目的README.md，而不是hope主题默认的页面(与本地部署后打开http://localhost:8080/展示的不一样)

**Reply by theCodedMike** [reply]
URL被吞了，这个"https://<USERNAME>.github.io/"

**Reply by theCodedMike** [reply]
https://USERNAME.github.io/

**Reply by theCodedMike** [reply]
解决了，"Oneupupup"这个老哥的回复是正确的

**Reply by Oneupupup** [reply]
很荣幸能解决你的问题

**Reply by zhiqinfff** [reply]
请问，改了branch为gh-pages，但是后面还有一个路径，但是选择哪个都不对，这个怎么解决呢

**Reply by zhiqinfff** [reply]
已解决，原因是很早之前我自定义过域名，而现在域名没有设置自定义导致的，总之跟部署无关
